### PR TITLE
Add launchpad (Closes #133)

### LINKED ISSUE #133: Add Launchpad integration to the plugin
Related to the work from Cyrille on RocketCDN plugin https://github.com/wp-media/rocketcdn-plugin/pull/33

### PR DESCRIPTION
**Pull request by nicomollet**
Closes #133 
RLL adapted to Launchpad structure

- Options save correctly
- RLL in effect with selected options, on images and iframes

### CODE REVIEW

**Comment by nicomollet** [comment]
@CrochetFeve0251 Can you tell why the framework looks loaded twice?
I see the admin menu page twice

image

**Comment by CrochetFeve0251** [comment]
> @CrochetFeve0251 Can you tell why the framework looks loaded twice? I see the admin menu page twice
> 
> image

@nicomollet the issue with the above code is that you included a subscriber twice.

**Comment by nicomollet** [comment]
Sure I will look into it @CrochetFeve0251

File: src/Admin/AdminPage.php [hunk 1]
```diff
@@ -58,22 +41,19 @@ class AdminPage
      *
      * @var string
      */
-    private $template_path;
+    private string $template_path;
```
Comment (CrochetFeve0251) [inline src/Admin/AdminPage.php]: @nicomollet I think this typing wasn't available in php 7.3

File: src/Subscriber/AdminPageSubscriber.php [hunk 2]
```diff
@@ -31,7 +31,7 @@ class AdminPageSubscriber implements SubscriberInterface {
 	 *
 	 * @var string
 	 */
-	private $plugin_basename;
+	private static string $plugin_basename;
```
Comment (CrochetFeve0251) [inline src/Subscriber/AdminPageSubscriber.php]: Same here for the type

File: src/EventManagement/SubscriberInterface.php [hunk 3]
```diff
@@ -35,5 +36,5 @@ interface SubscriberInterface
      *
      * @return array
      */
-    public function getSubscribedEvents();
+    public static function get_subscribed_events();
```
Comment (CrochetFeve0251) [inline src/EventManagement/SubscriberInterface.php]: Maybe https://github.com/wp-launchpad/core/blob/develop/inc/EventManagement/ClassicSubscriberInterface.php is more appropriate for this use case?

File: src/Subscriber/AdminPageSubscriber.php [hunk 4]
```diff
@@ -44,19 +44,19 @@ class AdminPageSubscriber implements SubscriberInterface {
...
 	 *
 	 * @return array
 	 */
-	public function getSubscribedEvents() {
+	public static function get_subscribed_events(): array {
 		return [
 			'admin_init'                                   => 'configure',
 			'admin_menu'                                   => 'addAdminPage',
-			"plugin_action_links_{$this->plugin_basename}" => 'addPluginPageLink',
+			"plugin_action_links_".self::$plugin_basename => 'addPluginPageLink',
```
Comment (CrochetFeve0251) [inline src/Subscriber/AdminPageSubscriber.php]: I am not sure this is working

**Review by CrochetFeve0251 — CHANGES_REQUESTED** [review]

**Comment by nicomollet** [comment]
✅ Tested with PHP:
- `7.3`
- `7.4`
- `8.0`
- `8.1`
- `8.2`

**Review by CrochetFeve0251 — APPROVED** [review]

**Pull request merged**